### PR TITLE
Styling/css general

### LINKED ISSUE #31: [MVP Comp] Review and finalize testing

### PR DESCRIPTION
**Pull request by jalbe0076**
This PR will close #31 
## What I did:
- Added conditionals on the albums details page to handle unwanted data
- Adjusted styling and responsiveness

## Testing Notes/Extras: 
- Visit each route and check the responsiveness

Did this break anything?

- [X]  No
- [ ]  Yes

Type of change

- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [X]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [X]  Skip all the other stuff and briefly explain the fix.
  - Styling

Checklist:

- [ ]  If this code needs to be tested, all tests are passing
- [X]  The code runs locally
- [ ]  There are comments where clarification/ organization is needed
- [X]  The code is DRY. If it's not, I tried my best
- [X]  I reviewed my code before pushing

What's next?
- TBD

### CODE REVIEW

File: src/components/Nav/Nav.js [hunk 1]
```diff
@@ -1,19 +1,20 @@
 import './Nav.css';
-import { NavLink, Link, Navigate, useNavigate } from 'react-router-dom';
+import { NavLink, Link, useLocation, useNavigate } from 'react-router-dom';
```
Comment (jmartin777) [inline src/components/Nav/Nav.js:2]: great use/update of 'useLocation' Hook here - allowing the navigation elements to change based on the user's authentication status and the current URL path.

File: src/api.js [hunk 2]
```diff
@@ -71,7 +71,7 @@ export function getAlbumDetails(albumID) {
         genre: data.genres,
         styles: data.styles,
         tracklist: data.tracklist?.map(track =>{
-          return `${track.position} ${track.title} ${track.duration}`  
+          return track.type_ === 'track' && `${track.position} ${track.title} ${track.duration}`
```
Comment (jmartin777) [inline src/api.js]: appreciate the conditional implemented here

File: src/components/Form/Form.js [hunk 3]
```diff
@@ -40,48 +40,52 @@ function Form ({title, artist, coverImg, id, showModal}) {
...
+                    }
+                  }}
+                placeholder={formattedDate}/>
+                <img className='calendar-icon'  src={process.env.PUBLIC_URL + "/images/calendar.png"} alt='calendar-icon'
+              />
+            </div>
           </div>
+          <img className='form-escape' src={process.env.PUBLIC_URL + "/images/escape-white.png"} alt='escape-icon' onClick={showModal} tabIndex={1}/>
         </div>
-        <img className='form-escape' src={process.env.PUBLIC_URL + "/images/escape-white.png"} alt='escape-icon' onClick={showModal}/>
```
Comment (jmartin777) [inline src/components/Form/Form.js:67]: what is form-escape do as functionality within the modal?

**Review by jmartin777 — APPROVED** [review]
Overall great job on the CSS update adjustments for everything! If i see anything else that comes up with CSS styling - ill let you know.

File: src/components/Journal/Journal.js [hunk 4]
```diff
@@ -15,11 +14,16 @@ const Journal = () => {
   }, [])
 
   return (
-    <div className='journal'>
-      {journal.length ?
-      journal.sort((a, b) => new Date(b.date) - new Date(a.date)).map(entry => <Entry key={entry.id} {...entry}/>)
-      : <p className='no-entries'>Search an album to add to your journal</p>}
```
Comment (mbenfowler) [inline src/components/Journal/Journal.js:21]: it looks like removing the `no-entries` p-tag is causing the test to fail - did you intend to remove this in your edit?

File: src/components/Collections/Collections.js [hunk 5]
```diff
@@ -18,9 +18,12 @@ const Collections = () => {
   });
 
   return (
+    <>
+    <h2 className='sub-title'>Collections</h2>
     <div className='album--grid'>
-      {savedAlbums.length ? savedAlbums : <h2>No albums in collection</h2>}
+      {savedAlbums.length ? savedAlbums : <h3>No albums in collection</h3>}
```
Comment (mbenfowler) [inline src/components/Collections/Collections.js:24]: you need to update this test as well to target `h3`

**Review by mbenfowler — CHANGES_REQUESTED** [review]
Just a couple ?s and a couple tests needing updated

File: src/components/Form/Form.js (hunk 3 above)
Comment (jalbe0076) [reply src/components/Form/Form.js:67]: It's there to style the exit button

**Comment by mbenfowler** [comment]
has there been any discussion yet to maybe line up the Nav component contents with the rest of the page when on landscape breakpoints?
Screenshot 2023-08-11 at 4 08 37 PM

File: src/api.js [hunk 6]
```diff
@@ -71,7 +71,7 @@ export function getAlbumDetails(albumID) {
         genre: data.genres,
         styles: data.styles,
         tracklist: data.tracklist?.map(track =>{
-          return `${track.position} ${track.title} ${track.duration}`  
+          return track.type_ === 'track' && `${track.position} - ${track.title} ${track.duration &&` - ${track.duration}`}`
```
[the remark this answers is not in the document]
Comment (jalbe0076) [reply src/api.js:74]: It's the key we get from Discogs

File: src/components/Journal/Journal.js (hunk 4 above)
Comment (jalbe0076) [reply src/components/Journal/Journal.js:21]: Yes, to match the other headers for the same conditional test. I'll adjust the test

**Comment by jalbe0076** [comment]
> has there been any discussion yet to maybe line up the Nav component contents with the rest of the page when on landscape breakpoints? Screenshot 2023-08-11 at 4 08 37 PM

You talking about removing the margin on the right from the title and left of the login so that it's lined up with the search bar that does not continue growing?

**Comment by mbenfowler** [comment]
no i think it just bugs me that the nav tabs are slight off center lol

**Comment by jalbe0076** [comment]
> no i think it just bugs me that the nav tabs are slight off center lol

Ah ok. Let's talk about this in our meeting.

File: src/components/Nav/Nav.css [hunk 7]
```diff
@@ -52,14 +54,25 @@
...
+}
+
+ h1:hover, .logout-btn:hover {
   color: #6546C4;
   filter: invert(31%) sepia(85%) saturate(1049%) hue-rotate(229deg) brightness(86%) contrast(97%);
 }
 
+.nav:hover {
+  color: #6546C4;
+  border: solid 1px #6546C4;
```
Comment (mbenfowler) [inline src/components/Nav/Nav.css:73]: ok a decided i don't like this sorry

File: src/components/Nav/Nav.css [hunk 8]
```diff
@@ -73,6 +86,8 @@
     justify-self: flex-start;
     text-align: center;
     margin-top: .7em;
+    position: relative;
+    right: auto;
```
Comment (mbenfowler) [inline src/components/Nav/Nav.css:90]: nice

**Review by mbenfowler — CHANGES_REQUESTED** [review]

**Review by mbenfowler — APPROVED** [review]

**Pull request merged**